var/spack/repos/builtin/packages/py-distributed/package.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyDistributed(PythonPackage):
+    """Distributed scheduler for Dask"""
+
+    homepage = "https://distributed.dask.org/"
+    url      = "https://pypi.io/packages/source/d/distributed/distributed-2.10.0.tar.gz"
+
+    version('2.10.0', '300e5495d0306f472c1aa34a522273e3066b9f50d3eb25cee17cd3b6177eea54')
+    version('1.28.1', sha256='3bd83f8b7eb5938af5f2be91ccff8984630713f36f8f66097e531a63f141c48a')
+
+    depends_on('python@2.7:2.8,3.5:', when='@:1')
+    depends_on('python@3.5:', when='@2:')
+    depends_on('py-setuptools', type='build')
```

Comment:
```suggestion
    depends_on('py-setuptools', type=('build', 'run'))
```

Reply:
Why? In general, how to tell if setuptools are needed at runtime?

Reply:
I _assume_ it's required at run-time because it's listed in `install_requires`. If it were only needed at build-time, it would usually be in `setup_requires` or simply imported in `setup.py`.

I _know_ it's requires at run-time because it's imported in `distributed/comm/registry.py` and a few other places. `setuptools` provides both `setuptools` and `pkg_resources`.